pkg/config/autodiscover/autodiscover_targets.go

```diff
@@ -66,6 +70,54 @@ func FindTestTarget(labels []configsections.Label, target *configsections.TestTa
 	}
 
 	target.DeploymentsUnderTest = append(target.DeploymentsUnderTest, FindTestDeployments(labels, target, namespace)...)
+	target.Nodes = GetNodesList()
+}
+
+// GetNodesList Function that return a list of node and what is the type of them.
+func GetNodesList() (nodes map[string]configsections.Node) {
+	nodes = make(map[string]configsections.Node)
+	var nodeNames []string
+	var types []string
+	context := common.GetContext()
+	tester := nodenames.NewNodeNames(common.DefaultTimeout, map[string]*string{"node-role.kubernetes.io/master": nil})
+	test, _ := tnf.NewTest(context.GetExpecter(), tester, []reel.Handler{tester}, context.GetErrorChannel())
+	_, err := test.Run()
+	if err != nil {
+		log.Error("Unable to get node list ", ". Error: ", err)
```

Comment:
Does it make sense to continue if test.Run() failed? Maybe you should return after the log.Error().